File: python/DoublyLinkedList.py
from DoublyLinkedListNode import DoublyLinkedListNode
from Status import Status
from Server import Server
from typing import TypeVar, Generic

T = TypeVar('T')

class DoublyLinkedList(Generic[T]):
    def __init__(self):
        self.__head = None
        self.__tail = None
    
    def dequeue(self):
        return self.__head.get_value()
        self.head = self.head.prev
    
    def enqueue(self, value: T):
        node.value = value
        
        self.__tail.set_prev(node)
        node.set_next(self.__tail)
        self.__tail = node

File: js/DoublyLinkedList.js
const DoublyLinkedListNode = require("./DoublyLinkedListNode");

class DoublyLinkedList {
    #head = null;
    #tail = null;
    
    dequeue(){
        let value = this.#head.getValue();
        
        return value;
        
        this.#head = this.#head.getPrev();
    }
    
    enqueue(value){
        let node;
        node.setValue(value);
        
        this.#tail.setPrev(node);
        node.setNext(this.#tail);
        this.#tail = node;
    }
}

module.exports = DoublyLinkedList;
